charts/rcon-web-admin/templates/deployment.yaml

```diff
@@ -0,0 +1,126 @@
+        {{- include "rcon-web-admin.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "rcon-web-admin.fullname" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: RWA_USERNAME
+              value: {{ .Values.rconWeb.username | default "admin" | quote }}
+            - name: RWA_PASSWORD
+              valueFrom: 
+                secretKeyRef:
+                  name: {{ .Values.rconWeb.passwordExistingSecret | default (include "rcon-web-admin.fullname" .) }}
+                  key: {{ .Values.rconWeb.passwordKey | default "password" }}
+            - name: RWA_ADMIN
+              value: {{ .Values.rconWeb.isAdmin | default false | ternary "TRUE" "FALSE" | quote }}
+            - name: RWA_RCON_HOST
+              value: {{ .Values.rconWeb.rconHost | default "127.0.0.1" | quote }}
+            - name: RWA_RCON_PORT
+              value: {{ .Values.rconWeb.rconPort | default "25575" | quote }}
+            - name: RWA_GAME
+              value: {{ .Values.rconWeb.game | default "minecraft" | quote }}
+            - name: RWA_SERVER_NAME
+              value: {{ .Values.rconWeb.serverName | default .Values.rconWeb.game | quote }}
+            - name: RWA_RCON_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.rconWeb.rconPasswordExistingSecret | default (include "rcon-web-admin.fullname" .) }}
+                  key: {{ .Values.rconWeb.rconPasswordKey | default "rcon-password" }}
+            - name: RWA_RESTRICT_COMMANDS
+              value: {{ join "," .Values.rconWeb.restrictCommands | quote }}
+            - name: RWA_RESTRICT_WIDGETS
+              value: {{ join "," .Values.rconWeb.restrictWidgets | quote }}
+            - name: RWA_READ_ONLY_WIDGET_OPTIONS
+              value: {{ .Values.rconWeb.immutableWidgetOptions | default false | ternary "TRUE" "FALSE" | quote }}
+            - name: RWA_WEB_RCON
+              value: {{ .Values.rconWeb.websocketRcon | default false | ternary "TRUE" "FALSE" | quote }}
+            {{- if .Values.ingress.enabled }}
+            {{ $wsUrl := print .Values.ingress.host (trimSuffix "/" .Values.ingress.path) "/websocket" }}
+            - name: RWA_WEBSOCKET_URL_SSL
+              value: {{ print "wss://" $wsUrl | quote }}
+            - name: RWA_WEBSOCKET_URL
+              value: {{ print "ws://" $wsUrl | quote }}
+            {{- end }}
+          {{- if not .Values.ingress.enabled }}
+          command:
```

Comment:
I don't really like this whole part, but there is unfortunate loop of "To deploy this chart you need to configure websocket url, to get websocket url you need to deploy this chart". Better solution would be to resolve it upstream, but for now this should work fine

Reply:
Understood and agreed that I don't like how rcon-web handles websockets in the first place with the requirement of a dedicated port.